### PR TITLE
work: proper timeout for new and rescheduled work

### PR DESCRIPTION
**Pull request by tlauda**
Takes work flag into consideration during initial and
rescheduled timeout setup.

Signed-off-by: Tomasz Lauda <tomasz.lauda@linux.intel.com>

### CODE REVIEW

**Comment by xiulipan** [comment]
@tlauda 
So you fixed some work queue lost issue?
Good catch.

**Comment by zrombel** [comment]
@tlauda disk space problem is done, but there is still problem with Klocwork on QB and we are still working on it.

**Comment by lgirdwood** [comment]
@tlauda can you force push again to redo CI then we can merge if all is OK.

**Comment by tlauda** [comment]
@lgirdwood Done, but the CI failed and I have no idea why.
@xiulipan Any ideas?

**Comment by lgirdwood** [comment]
@tlauda Travis boot passed, I've no idea why Jenkins filing.

**Comment by xiulipan** [comment]
@lgirdwood @tlauda 
Retry with local build
It seems the agent has timeout of unknown reason.
CHT will 100% panic and BYT will random 10% panic in my local.
Re triggered travis-ci and got a byt failed
https://travis-ci.org/thesofproject/sof/jobs/483369689

Detail with https://github.com/thesofproject/sof/issues/874

**Comment by xiulipan** [comment]
@lgirdwood 
Random regression happen on BYT and CHT
Maybe revert this PR first and figure out if we have wrong with agent or work queue.

File: src/lib/work.c [hunk 1]
```diff
@@ -379,8 +379,13 @@ void work_schedule(struct work_queue *queue, struct work *w, uint64_t timeout)
 	}
 
 	/* convert timeout micro seconds to CPU clock ticks */
-	w->timeout = queue->ticks_per_msec * timeout / 1000 +
-		work_get_timer(queue);
+	if (w->flags & WORK_SYNC)
+		w->timeout =
+			queue_calc_next_timeout(queue, work_get_timer(queue));
```
Comment (xiulipan) [inline src/lib/work.c]: @tlauda 
How do we use the ` uint64_t timeout` in L364 with you changes here?

**Pull request merged**